contracts/USM.sol

```diff
@@ -72,11 +72,21 @@ contract USM is IUSM, Oracle, ERC20Permit, WithOptOut, Delegable {
         WithOptOut(optedOut_)
     {
         oracle = oracle_;
-        fum = new FUM(this, optedOut_);
+        FUM fum_ = fum = new FUM(this, optedOut_);
+        addDelegate(address(fum_));
```

Comment:
Do you need this *and* the `msg.sender == address(fum)` condition below?

Reply:
The constructor needs to be reverted, sorry.